### PR TITLE
[OBSDEF-12159] add pravega/zk security context

### PR DESCRIPTION
**Pull request by realAaronWu**
## Purpose
[OBSDEF-12159](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-12159)
Add securoty context for pravega and zk pods, and init contianer for zk pods

## PR checklist
- [x] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [x] vSphere7 make package deployments passed
- [x] helm upgrade <chart> passed
- [x] helm test <release_name> passed
- [x] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
_Paste URL of charts-custom-ci job here_
https://asd-ecs-jenkins.isus.emc.com/view/ECS-Flex/job/charts-custom-ci/399/
_Paste the output of your helm install/vSphere7 deployment testing here_

https://asd-ecs-jenkins.isus.emc.com/job/stream-master-ci-vsph7-large/105/

### CODE REVIEW

File: ecs-cluster/templates/ecs-cluster.yaml [hunk 1]
```diff
@@ -226,6 +229,19 @@ spec:
...
+      - sh
+      - -c
+      - chown -R 1001:1000 /data /opt /zk-sp.sh
+      image: {{ .Values.global.registry }}/{{ .Values.zookeeper.image.repository }}:{{ .Values.zookeeper.image.tag }}
+      name: initscripts
+      resources: {}
+      securityContext:
+        runAsUser: 0
+      volumeMounts:
+      - mountPath: /data
```
Comment (eddiepavkovic) [inline ecs-cluster/templates/ecs-cluster.yaml:243]: This should be done in a Dockerfile.  not at runtime.

@Bjohnson131 @vgupta-mickey

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]

File: ecs-cluster/templates/ecs-cluster.yaml (hunk 1 above)
Comment (realAaronWu) [reply ecs-cluster/templates/ecs-cluster.yaml:243]: I tried to add permssion in Dockfile but it failed to take effect when contianer starts, the zk still got permission denied error when access /data.

Comment (realAaronWu) [reply ecs-cluster/templates/ecs-cluster.yaml]: @eddiepavkovic  removed the initContainer and use fsGroup to set permission instead.

**Review by BassinD — APPROVED** [review]

**Review by eddiepavkovic — APPROVED** [review]
g2g

**Pull request merged**